homeassistant/components/apsystems_local/sensor.py

```diff
@@ -0,0 +1,166 @@
+        translation_key="total_power_p1",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="total_power_p2",
+        translation_key="total_power_p2",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.p2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production",
+        translation_key="lifetime_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.te1 + c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p1",
+        translation_key="lifetime_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.te1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="lifetime_production_p2",
+        translation_key="lifetime_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.te2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production",
+        translation_key="today_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1 + c.e2,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p1",
+        translation_key="today_production_p1",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e1,
+    ),
+    ApsystemsLocalApiSensorDescription(
+        key="today_production_p2",
+        translation_key="today_production_p2",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = config["COORDINATOR"]
+    device_name = config[CONF_NAME]
```

Comment:
get the name from the config entry. That way if a user renames config entry, it will rename their device 👍